common.py

```diff
@@ -1230,46 +1317,33 @@ def split_bdt(sel):
 
 def calc_bdt_scores(X, model_file=bdt_model_file):
     import xgboost as xgb
-
     # Load the model and get the predictions
     xgb_model = xgb.XGBClassifier()
     xgb_model.load_model(model_file)
     with time_and_log(f'Calculating xgboost scores for {bdt_model_file}...'):
         score = xgb_model.predict_proba(X)[:,1]
+    del xgb_model
     return score
 
+
 DDT_PATH_BDTBASED='root://cmseos.fnal.gov//store/user/lpcdarkqcd/boosted/BDT_based/'
-DDT_FILE_BDTBASED='models/bdt_ddt_AN_v6.json'
+DDT_FILE_BDTBASED='models/bdt_ddt_map_ANv6_3d.json'
+DDT_FILE_BDTBASED_RT_DDT='models/bdt_ddt_withRT_ANv6_3d.json'
 
 def apply_bdtbased(cols,wp,lumi,anti=False,model_file=bdt_model_file,ddt_map_file=DDT_FILE_BDTBASED, xrootd_url=DDT_PATH_BDTBASED):
     check_if_model_exists(ddt_map_file, xrootd_url)
     check_if_model_exists(model_file, xrootd_url)
-    cols = apply_rt_signalregion(cols)
-    # calc_bdt_scores doesn't like empty inputs
+
+    cols = apply_rt_signalregion(cols) if ddt_map_file == DDT_FILE_BDTBASED else apply_rt_signalregion_ddt(cols)
```

Comment:
this check is a little fragile. but the only way to have a more general guarantee would be to promote the DDT file from a string to a class with additional attributes, which is maybe an unnecessary amount of work at this juncture.